Sources/HummingbirdHTTP2/HTTP2Channel.swift

```diff
@@ -0,0 +1,109 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2023 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Logging
+import NIOCore
+import NIOPosix
+import NIOHTTP1
+import NIOHTTP2
+import Hummingbird
+import HummingbirdCore
+import NIOSSL
+
+public struct HTTP2Channel: HTTPChannelHandler {
+    public typealias Value = EventLoopFuture<NIONegotiatedHTTPVersion<HTTP1Channel.Value, (NIOAsyncChannel<HTTP2Frame, HTTP2Frame>, NIOHTTP2Handler.AsyncStreamMultiplexer<HTTP1Channel.Value>)>>
+
+    private let sslContext: NIOSSLContext
+    private var http1: HTTP1Channel
+    private let additionalChannelHandlers: @Sendable () -> [any RemovableChannelHandler]
+    public var responder: @Sendable (HBHTTPRequest, Channel) async throws -> HBHTTPResponse {
+        get { http1.responder }
+        set { http1.responder = newValue }
+    }
+
+    public init(
+        tlsConfiguration: TLSConfiguration,
+        additionalChannelHandlers: @autoclosure @escaping @Sendable () -> [any RemovableChannelHandler] = [],
+        responder: @escaping @Sendable (HBHTTPRequest, Channel) async throws -> HBHTTPResponse = { _, _ in throw HBHTTPError(.notImplemented) }
+    ) throws {
+        var tlsConfiguration = tlsConfiguration
+        tlsConfiguration.applicationProtocols = NIOHTTP2SupportedALPNProtocols
+        self.sslContext = try NIOSSLContext(configuration: tlsConfiguration)
+        self.additionalChannelHandlers = additionalChannelHandlers
+        self.http1 = HTTP1Channel(additionalChannelHandlers: additionalChannelHandlers(), responder: responder)
+        self.responder = responder
```

Comment:
Don't need to set `self.responder` here as it is already set in the creation of `self.http1`

Reply:
Good catch